src/base/emu-i386/simx86/codegen-x86.c

```diff
@@ -1239,35 +1239,13 @@ arith0:		{
 			0x66,0x09,0xd0,
 		};
 		unsigned char *q=Cp; GNX(Cp, pseqpre, sizeof(pseqpre));
-		if (mode&DATA16) q[8] = 0xfe; /* use -2 in lea ins */
-//		if (in_dpmi) {
-		    /* This solves the DOSX 'System test 8' error.
-		     * The virtualized IF is pushed instead of the
-		     * real one (which is always 1). This way, tests
-		     * on the pushed value of the form
-		     *		cli
-		     *		pushf
-		     *		test 0x200,(esp)
-		     * don't fail anymore. It is not clear, apart this
-		     * special test case, whether pushing the virtual
-		     * IF is actually useful; probably not. In any
-		     * case, POPF ignores this IF on stack.
-		     * Since PUSHF doesn't trap in PM, non-cpuemued
-		     * dosemu will always fail this particular test.
-		     */
-			// rcr $10,%%eax	(IF->cy)
-			G3M(0xc1,0xd8,0x0a,Cp);
-			// bt $19,(_EFLAGS-TheCPU)(%ebx) (test for VIF)
```

Comment:
Maybe at least leave that code
disabled, as a reference for someone
implementing VME later?
By removing it, you reduce the chances
of re-impl to near zero.

Reply:
Actually could you please just
make it selectable with ifdef?

Reply:
Or is it not so easy?

Reply:
I could ifdef it but it needs some more changes to be enabled of course, also then we have different code for PM than for V86.

Alternatively I could implement it as an interpreted branch in interp.c for IOPL<3.
Similarly I can re-add the other deleted sections in interp.c with a guard of IOPL<3. I'll do that tomorrow then.

Reply:
Not actually sure if its necessary
to move it to interp, because its a
pessimization.
I'd rather put it under a different
primitive in codegen, so that the
interp can just choose what to
generate. But since this require
more work, just if0 would be fine
(w/o taking any care for it to work).